sql/civicrm_option_groups/acl_role.sqldata.php

```diff
@@ -0,0 +1,9 @@
+<?php
+return CRM_Core_CodeGen_OptionGroup::create('acl_role', 'a/0008')
+  ->addMetadata([
+    'title' => ts('ACL Role'),
+  ])
+  ->addValues(['label', 'name', 'value'], [
+    [ts('Administrator'), 'Admin', 1],
+    [ts('Authenticated'), 'Auth', 2, 'is_reserved' => 1],
```

Comment:
IMO this pattern of splitting the keys and values into 2 different arrays is not very dev friendly, especially for long lists with lots of columns. It's not as bad as the sql.tpl file because it's shorter, but it's still basically the same readability problem where the header is in one spot and the data is elsewhere and you have to scroll and put your finger on the screen and count columns to see what's going on. The fact that you can randomly mix in associative keys into the otherwise non-associative values array feels extra-confusing; it all just seems like an accident waiting to happen. I would vote to change the signature of `addValues` to only accept a single param, and use our standard code formatting instead of squashing values into a single row, like this:

```suggestion
  ->addValues([
    [
      'label' => ts('Administrator'),
      'name' => 'Admin',
      'value' => 1,
    ],
    [
      'label' => ts('Authenticated'),
      'name' => 'Auth',
      'value' => 2,
      'is_reserved' => 1,
    ],
```

Reply:
It's also matches the output from the api explorer, which could be convenient.

Reply:
Yeah, there are some example where I agree... and others where I disagree...

> especially for long lists with lots of columns

Maybe that's the thing. It's an option-list. You really only need 1-2 columns. At a stretch, 3-4 columns for some quirky cases. A large percentage of OGs/OVs fit very nicely into table layout, as long you focus on the columns that are actually used by that OG.

But of course, `civicrm_option_value` is more complicated than that -- it's ~15 columns. So anytime you deal with an OG, you have to figure out: "Which columns are allowed or required in this OG? Are there oddball records with unusual/asymmetric fields?" (*in addition to the other obvious questions, like "What does `value` mean?"*)

And so there are several OG's with really complex data-structures, and the key-value-tree is easier to read. But there are also several cases that are simple -- where the simple-pairs are easier to read. With ~90 OG's, there are a lot of different situations. And pragmatically, the work to convert varies by group (*some have quirkier mappings that needed hand-tuning*).

So... cost-benefit varies a bit. I guess I'm cutting the baby in half... I'm in the process of converting a large batch where it seems useful or easy.

Reply:
So you are saying - yeah it would be easier to read if it were keyed but this is already a lot of work & I don't want to add to it?

Reply:
@eileenmcnaughton Sort of, not exactly. For me, the formats are not intrinsically better or worse - it depends on the data being described. 

* [label_format](https://github.com/civicrm/civicrm-core/blob/eb0359fb06da07697e71fcbfac696638f3fcebba/sql/civicrm_option_groups/label_format.sqldata.php) and [email_greeting](https://github.com/civicrm/civicrm-core/blob/eb0359fb06da07697e71fcbfac696638f3fcebba/sql/civicrm_option_groups/email_greeting.sqldata.php) are clearly better with spacious records (to my eye).
*  [mobile_provider](https://github.com/civicrm/civicrm-core/blob/eb0359fb06da07697e71fcbfac696638f3fcebba/sql/civicrm_option_groups/mobile_provider.sqldata.php) and [safe_file_extension](https://github.com/civicrm/civicrm-core/blob/eb0359fb06da07697e71fcbfac696638f3fcebba/sql/civicrm_option_groups/safe_file_extension.sqldata.php) are clearly better with tidy lists (to my eye).
* [environment](https://github.com/civicrm/civicrm-core/blob/eb0359fb06da07697e71fcbfac696638f3fcebba/sql/civicrm_option_groups/environment.sqldata.php) and [contact_edit_options](https://github.com/civicrm/civicrm-core/blob/eb0359fb06da07697e71fcbfac696638f3fcebba/sql/civicrm_option_groups/contact_edit_options.sqldata.php) are middling. (In either format, but they're acceptable but they're not models of elegance.)

Pragmatically, with ~90 of them, it's messy to give nice rationalizations to serve them all. I just did a couple passes of making judgment calls (swayed by a mix of personal aesthetics and laziness).